### PR TITLE
Feature/sql worker

### PR DESCRIPTION
**Pull request by VaclavK**
#40475

sample output schema including the schema table

image

### CODE REVIEW

File: src/SapAct/Models/Metrics/SapActProcessingMetric.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿
+namespace SapAct.Models.Metrics
+{
+	internal class SapActProcessingMetric
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/SapAct/Models/Metrics/SapActProcessingMetric.cs:4]: ah thanks for spotting

File: src/SapAct/Services/SQLService.cs [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+									//schema was updated by another instance but let's check against persistent storage
+									var status = await CheckObjectTypeSchemaAsync(objectType!, dataVersion!, TargetStorageEnum.SQL);
+									updateNecessary = status != SchemaCheckResultState.Current;
+								}
+							} while (updateNecessary);
+						}
+
+
+						await SinkDataAsyncInner(payload, schemaDescriptor, new KeyDescriptor { RootKey = objectKey, ForeignKey = string.Empty });
+						await sqlTransaction.CommitAsync();
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/SapAct/Services/SQLService.cs:59]: I intentionally do not do this... I let each message to process , cancellation token is then applied at next message being peeked

File: src/SapAct/Services/SQLService.cs [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+		}
+		catch (Exception ex)
+		{
+			logger.LogError(ex, "Error upserting SQL structures");
+
+			transaction.Rollback();
+			throw;
+		}
+
+		transaction = null;
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/SapAct/Services/SQLService.cs:192]: I am not sure why I did that.. removing as it indeed seems unnecessary

File: src/SapAct/Services/SQLService.cs [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+		bool addComma = false;
+		tableUpdateSB.AppendLine($"ALTER TABLE {tableName} ADD");
+		foreach (var column in addedColumns)
+		{
+			if (addComma)
+				tableUpdateSB.Append(',');
+
+			tableUpdateSB.AppendLine($"{column} NVARCHAR(255) NULL");
+			addComma = true;
+		}
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/SapAct/Services/SQLService.cs]: dah... yeah.. thanks :D

**Review by yavar-hasanov — APPROVED** [review]

**Pull request merged**